s101.js

```diff
@@ -155,7 +155,7 @@ S101Codec.prototype.handleEmberFrame = function (frame) {
   }
 
   var payload = frame.readBuffer()
-  payload = payload.slice(0, payload.length - 2)
+  payload = payload.slice(2, payload.length - 2)
```

Comment:
Do you have a sniffer trace that triggers the problem ?
I have a a device in my lab using mutli frames and this code works fine.

Reply:
```
00000000  fe 00 0e 00 01 c0 01 02  1f 02 60 10 6b 0e a0 0c   ........ ..`.k...
00000010  62 0a a0 03 02 01 20 a1  03 02 01 fd df 94 8f ff   b..... . ........
    00000000  fe 00 0e 00 01 c0 01 02  14 02 60 80 6b 80 a0 80   ........ ..`.k...
    00000010  63 80 a0 03 02 01 01 a1  80 31 80 a0 0a 0c 08 53   c....... .1.....S
    00000020  61 70 70 68 69 72 65 00  00 00 00 00 00 00 00 00   apphire. ........
    00000030  00 00 00 22 24 ff                                  ..."$.
```

Reply:
The parsed payload going in to `handleEmberFrame()` is `<Buffer 14 02 60 80 6b 80 a0 80 63 80 a0 03 02 01 01 a1 80 31 80 a0 0a 0c 08 53 61 70 70 68 69 72 65 00 00 00 00 00 00 00 00 00 00 00 00 22 24>`

The error says

` 0x60: got 0x14`

Stripping out 14 02 from the buffer, reading from 60, solves my problem.

Reply:
This is a single-packet message btw.

Reply:
`0x14 0x02` is DTD version minor and major, meaning v2.20?

I'm gonna dive deeper looking for how to handle this.

Reply:
That's it. We're never reading or parsing the DTD minor/major bytes. They are present in the encoder, but never read/handled in the encoder.

Do you have any input on this, or can I just go ahead and handle them somehow? Strange the decoder works for you without handling this? Or are you only encoding?